provider/slof.py

```diff
@@ -0,0 +1,142 @@
+"""
+Module for providing common interface to test SLOF component.
+"""
+
+import re
+import time
+import logging
+import os
+
+from virttest import utils_misc
+
+
+def get_boot_content(vm, start_pos=0, start_str='SLOF',
+                     end_str='Successfully loaded'):
+    """
+    Get SLOF specified content by reading the serial log.
+    :param vm: VM object.
+    :param start_pos: int, start position which start to read.
+    :param start_str: str, start string.
+    :param end_str: str, end string.
+    :return: content list and next position.
+    """
+
+    def _catch_specify_content(cur_pos, string):
+        content = []
+        with open(vm.serial_console_log) as f:
+            for line in f.readlines()[cur_pos:]:
+                content.append(line)
+                if string in line:
+                    return content, cur_pos + 1
+                cur_pos += 1
+            return None, None
+
+    cur_pos = start_pos
+    with open(vm.serial_console_log) as f:
+        for line in f.readlines()[cur_pos:]:
+            if start_str in line:
+                return _catch_specify_content(cur_pos, end_str)
+            cur_pos += 1
+        return None, None
```

Comment:
is it better `return None, cur_pos` instead of `return None, None`, the returned `cur_pos` could be used for next call to reduce iterations.